### PR TITLE
DSA website

### PR DESCRIPTION
**Pull request by jtomeck**

### CODE REVIEW

**Comment by manthey** [comment]
We just merged the girder-3-deployment branch to master.  If you rebase this on current master the CI tests should pass.

**Comment by manthey** [comment]
When I build this locally via `bundle exec jekyll serve`, it throws a warning `Invalid theme folder: _sass`.

**Comment by jtomeck** [comment]
> When I build this locally via `bundle exec jekyll serve`, it throws a warning `Invalid theme folder: _sass`.

Yeah, I researched that a bit and it seems to be just a warning that does not effect anything. I do plan to put a readme in the docs folder eventually to explain the process of getting things running and installed.

**Comment by jtomeck** [comment]
@manthey @cooperlab - I saw you provided the collaborator logos, but I found this version online. Is it okay to use it, since it looks like a more modern version?

logo-feinberg_school_of_medicine

**Comment by faiza-a** [comment]
> @manthey @cooperlab - I saw you provided the collaborator logos, but I found this version online. Is it okay to use it, since it looks like a more modern version?
> 
> logo-feinberg_school_of_medicine

@manthey  @cooperlab  And here is the logos provided earlier.
https://drive.google.com/drive/folders/1CNR57m36TZ8_fzCF7u49UQ3xyQxYVhZt

**Comment by jtomeck** [comment]
> I am doing couple of minor changes before merging.

@faiza-a - I would wait to merge until we have everyone's buy-in. Currently I have it on a draft PR so that the two of us can collaborate and get it to a spot we're happy with. Once we're happy, I'll mark the PR ready for review and get everyone else's review.

**Comment by faiza-a** [comment]
@manthey the floodgates are open for feedback on the PR

**Comment by manthey** [comment]
Great.  Should I expect everything to work, or are some links expected to be missing yet (such as the link to the demo)?

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      - name: Feature Extraction and Predictive Modeling
+        description: Object and patch-level features describing shape, texture, and color can be used to build machine-learning models.
+        icon: assets/img/feature-icons/icon-feature_extraction.svg
+
+      - name: Extensibility
+        description: Users can integrate their custom algorithms through a containerization process that auto-generates DSA user-interfaces.
+        icon: assets/img/feature-icons/icon-extensibility.svg
+
+      - name: Broad Support for Histology Image Formats
+        description: A wide variety of tiled image formats are supported, including tiff, svs, and jp2. Images can be retiled automatically as needed for processing algorithms. Additional formats can be added with a pluggable Python interface.
```
Comment (manthey) [inline docs/index.md]: Let's change this side of the the image support format to read "The same wide variety of histology images that can be viewed can be used with any processing algorithms.  Sub-images can be processed at custom tile sizes and magnifications as needed."

**Comment by jtomeck** [comment]
> Great. Should I expect everything to work, or are some links expected to be missing yet (such as the link to the demo)?

So far this is just the homepage. No other pages exist, so that's something we'll need to be thinking about.

**Comment by faiza-a** [comment]
> > Great. Should I expect everything to work, or are some links expected to be missing yet (such as the link to the demo)?
> 
> So far this is just the homepage. No other pages exist, so that's something we'll need to be thinking about.

Yes
This is Landing Page design -first pass review
please review the content and general design layout,
- Logo (is the current logo fine or have any other ideas for it)
- Headings and text 
- buttons  (links are not working right now)
- any other change request/suggestion on webpage design so far

Also please provide/confirm links for these
**Gitter:**
**Discourse:**
**Youtube:**    (do we have a youtube channel or do i remove this button?)
**Jupyter Notebook:**
**Demo:**  http://demo.kitware.com/histomicstk/histomicstk
**DSA Github:**  https://github.com/DigitalSlideArchive
**DSA Docs:**  https://digitalslidearchive.github.io/HistomicsTK/
**Histomics UI Github:**  https://github.com/DigitalSlideArchive/HistomicsUI
**Histomics UI Docs:** 
**Histomics TK Github:**  https://github.com/DigitalSlideArchive/HistomicsTK
**Histomics TK  Docs:** 
**Demos and examples:**
**Success Stories:**
**Paper/Publications:**

File: docs/index.md (hunk 1 above)
Comment (faiza-a) [reply docs/index.md]: done

**Comment by manthey** [comment]
> Also please provide/confirm links for these

**Gitter:**  https://gitter.im/DigitalSlideArchive/HistomicsTK
**Discourse:** https://discourse.girder.org/c/histomicstk

> **Youtube:** (do we have a youtube link or do i remove this button?)

We have an old video that was posted on YouTube:
https://www.youtube.com/channel/UCJVNVQXivYPDb4goNU4tK3A

> **Jupyter Notebook:**

There are links to a variety of Jupyter notebooks here:
https://digitalslidearchive.github.io/HistomicsTK/examples.html

> **Demo:** http://demo.kitware.com/histomicstk/histomicstk
> **DSA Github:** https://github.com/DigitalSlideArchive

**DSA Docs:** https://github.com/DigitalSlideArchive/digital_slide_archive/blob/master/ansible/README.rst
Perhaps we should change this link to explicitly say "Installation" rather than Docs.

> **Histomics UI Github:** https://github.com/DigitalSlideArchive/HistomicsUI

**Histomics UI Docs:** https://github.com/DigitalSlideArchive/HistomicsUI/blob/master/README.rst

> **Histomics TK Github:** https://github.com/DigitalSlideArchive/HistomicsTK

**Histomics TK Docs:** https://digitalslidearchive.github.io/HistomicsTK/

> **Demos and examples:**
> **Success Stories:**
> **Paper/Publications:**

File: docs/_sass/_main.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
+@charset 'utf-8';
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&display=swap');
+
+$family-sans-serif: 'Open Sans', sans-serif;
+
+// Color Variables
+$primary: #2092d2 !default;
+$secondary: #2fb39b !default;
+$gray: #999;
```
Comment (manthey) [inline docs/_sass/_main.scss]: I find the contrast on the text in the "What is the DSA" section a bit low.  Can we use a somewhat darker grey here?

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  - link_text: Read Docs
+    link: #
+
+  - link_text: View on GitHub
+    link: #
+
+# What is DSA Section
+# =================================
+about_title: What is DSA?
+about_text: The Digital Slide Archive (DSA) is a platform  that provides the ability to store, manage, visualize and annotate large imaging data sets.  The DSA is an umbrella term and consists of an analysis toolkit (HistomicsTK), a library to manage/visualize slides (HistomicsUI) as well as a database layer (using Mongo), and a web-server that provides a rich API.  This system allows;
```
Comment (manthey) [inline docs/index.md]: I'd reword this somewhat and not end on a semicolon.  Perhaps: `The Digital Slide Archive (DSA) is a platform that provides the ability to store, manage, visualize and annotate large imaging data sets.  The DSA consists of an analysis toolkit (HistomicsTK), an interface to visualize slides and manage annotations (HistomicsUI), a database layer (using Mongo), and a web-server that provides a rich API and data management tools (using Girder).  This system can:`

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  - name: Youtube Channel
+    link: #
+    icon: assets/img/resource-icons/icon_youtube.png
+
+# Platforms Content
+# =================================
+platforms:
+  # HistomicsUI
+  - title: HistomicsUI
+    description: Web-based application for examining, annotating, and processing histology images to extract both low and high level features (e.g. cellular structure, feature types).
```
Comment (manthey) [inline docs/index.md]: Let's turn this into a complete sentence: `HistomicsUI is a web-based ...`

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        description: Girder provides distributed execution and monitoring of algorithm and analytics jobs.
+        icon: assets/img/feature-icons/icon-execution_engine.svg
+
+      - name: Broad Support for Histology Image Formats
+        description: A wide variety of tiled image formats are supported, including tiff, svs, and jp2. Images can be retiled automatically as needed for processing algorithms. Additional formats can be added with a pluggable Python interface.
+        icon: assets/img/feature-icons/icon-history_image_formats-primary.svg
+
+  # HistomicsTK
+  - title: HistomicsTK
+    description: HistomicsTK is a Python toolkit based on image processing, for quantitative analysis of whole-slide digital pathology images.
```
Comment (manthey) [inline docs/index.md]: I think this would sound better: `HistomicsTK is a Python image-processing toolkit for quantitative analysis of whole-slide digital pathology images.`

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    docs_url: google.com
+    github_url: github.com
+    screenshot: assets/img/histomicstk_screenshot.png
+    features:
+      - name: Preprocessing and Format Support
+        description: HistomicsTK provides color normalization and deconvolution operations to improve the robustness of analytic pipelines. The Large Image plugin gives programmers a convenient interface for reading from whole-slide imaging formats.
```
Comment (manthey) [inline docs/index.md]: Since we mention formats in greater detail below, lets change this section to just `Preprocessing` and remove the "The Large Image plugin..." sentence.

Comment (faiza-a) [reply docs/index.md]: fixed in commit 51f03b5f5d244cd83cc6acb83fbdc3cba4987b7d

File: docs/index.md (hunk 5 above)
Comment (faiza-a) [reply docs/index.md]: Fixed in 51f03b5f5d244cd83cc6acb83fbdc3cba4987b7d

File: docs/index.md (hunk 4 above)
Comment (faiza-a) [reply docs/index.md]: Fixed in 51f03b5f5d244cd83cc6acb83fbdc3cba4987b7d

File: docs/index.md (hunk 3 above)
Comment (faiza-a) [reply docs/index.md]: Fixed in 27f408b8d0311cdf44401bbdd023452f144ef19f

Comment (faiza-a) [reply docs/index.md]: Fixed in 3ca419c83a963bdafaf7e336b4cda927c524722b

File: docs/_sass/_main.scss (hunk 2 above)
Comment (faiza-a) [reply docs/_sass/_main.scss]: Fixed in 27f408b8d0311cdf44401bbdd023452f144ef19f

**Comment by manthey** [comment]
@cooperlab, @dgutman Feedback welcome.  As a reminder, you can see a preview on Circle-CI under the docs test's artifacts (though you may need to log in to do so).

**Comment by manthey** [comment]
@faiza-a I know we'll need to make additional pages for papers, demos, and stories.  I'd like to defer those to a future PR so we can merge this and have something up.  Aside from adding the links I listed in my previous comment and getting agreement with @dgutman and @cooperlab, what else do we need to merge this?

**Comment by faiza-a** [comment]
> @faiza-a I know we'll need to make additional pages for papers, demos, and stories. I'd like to defer those to a future PR so we can merge this and have something up. Aside from adding the links I listed in my previous comment and getting agreement with @dgutman and @cooperlab, what else do we need to merge this?


If we defer additional pages for next PR. I think from my side we are good to merge this PR after adding links. i will add the links and do final tweaks by tomorrow.

**Comment by faiza-a** [comment]
> > @faiza-a I know we'll need to make additional pages for papers, demos, and stories. I'd like to defer those to a future PR so we can merge this and have something up. Aside from adding the links I listed in my previous comment and getting agreement with @dgutman and @cooperlab, what else do we need to merge this?
> 
> If we defer additional pages for next PR. I think from my side we are good to merge this PR after adding links. i will add the links and do final tweaks by tomorrow.

@manthey 
- Added all links that you provided.
- Also added links on collaborators to their websites/profiles. (let me know if you want to change or remove that)
- Made some design changes to what is DSA section. (I couldn't find Jupyter notebooks logo icon. i will fix that later) 
- Have put 'coming soon' on 'success stories', 'papers/publications' and 'demos' for now until we make those pages. (is this fine. or you have any other idea for this?)

Let me know if there are any change requests on this PR, otherwise I think this is good for merging from my side. 

For the other pages for papers, demos and stories, where can i find the content?

**Comment by manthey** [comment]
Can we add a `mailto:kitware@kitware.com` link to the Contact button?

Should we temporarily disable the top bar Demos/Stories/Publications?

Where do we expect the About link to go?

I'll write up something for Demos, which will probably be annotated links to the demo on demo.kitware.com.  We can add links to Emory computers later.

I'm not sure what the stories are yet.

We have a list of publications.  @dgutman or @cooperlab, do either of you have the list in a handy spot?  If not, I'll pull it from a recent proposal we've done.

I'm not sure why it does it, but the preview of the website on Circle-CI does odd things to the absolute url links.  The actual deployment will be fine.  I'll check to see how to avoid this problem for artifact testing.

**Comment by faiza-a** [comment]
> Can we add a `mailto:kitware@kitware.com` link to the Contact button?
> 
> Should we temporarily disable the top bar Demos/Stories/Publications?
> 
> Where do we expect the About link to go?

Sorry i forgot about the header.
I have removed Demos/Stories/Publications buttons
Added contact - mailto and visit contact page buttons.
about and collaborator buttons scroll down to the about and collaborators sections.

**Comment by manthey** [comment]
Great.  The only remaining issue I have is that the preview in the Circle-CI artifacts tab prefixes absolute links.  When built locally (without a baseurl), it all works fine.

I think this is a Jekyll template issue, Most links in the `_include` templates are of the form `{{ item.link | prepend: site.baseurl }}` (this could also be written as `{{ item.link | relative_url }}`).  If they are absolute links, they should probably be for the form `{{ item.link }}`.  I don't know if it is worth changing this; it would make the preview in Circle-CI nicer.

Thoughts?

**Comment by faiza-a** [comment]
> Great. The only remaining issue I have is that the preview in the Circle-CI artifacts tab prefixes absolute links. When built locally (without a baseurl), it all works fine.
> 
> I think this is a Jekyll template issue, Most links in the `_include` templates are of the form `{{ item.link | prepend: site.baseurl }}` (this could also be written as `{{ item.link | relative_url }}`). If they are absolute links, they should probably be for the form `{{ item.link }}`. I don't know if it is worth changing this; it would make the preview in Circle-CI nicer.
> 
> Thoughts?

Hey (I was out of office yesterday). i will look into it and will fix this on next PR. 

> I'll write up something for Demos, which will probably be annotated links to the demo on demo.kitware.com. We can add links to Emory computers later.
> 
> I'm not sure what the stories are yet.
> 
> We have a list of publications. @dgutman or @cooperlab, do either of you have the list in a handy spot? If not, I'll pull it from a recent proposal we've done.

Alright, i will work on generating other web pages. and as i get the data i will keep entering it.

**Comment by manthey** [comment]
Posted here: http://digitalslidearchive.github.io/digital_slide_archive/

And, it looks like we need to fix those absolute links in the Github build, too.

**Comment by jtomeck** [comment]
> And, it looks like we need to fix those absolute links in the Github build, too.

Would it make sense to just get rid of the `| prepend` wherever it's used? I don't see a reason to have it in the first place, and it may just be that this is theme functionality that we do not need.

**Comment by manthey** [comment]
If the links are correctly relative, then it might be okay to just remove it.  We'd have to give it a try to make sure it works.  Based on what I see, if it works in the circle-ci artifact, it will work on the github.io page.

**Comment by jtomeck** [comment]
@manthey - I'll make a PR to test it, if it works we can merge it in.
